### PR TITLE
Fix bugs that showed up on Sentry

### PR DESCRIPTION
**Pull request by Dantemss**
https://sentry.cnx.org/openstax/biglearn-scheduler/issues/21280/
https://sentry.cnx.org/openstax/biglearn-scheduler/issues/21292/
https://sentry.cnx.org/openstax/biglearn-scheduler/issues/21314/

### CODE REVIEW

File: app/domain/services/fetch_course_events/service.rb [hunk 1]
```diff
@@ -131,21 +131,30 @@ def fetch_and_process_course_events(courses, event_types, restart, current_time)
       course_uuids_to_requery << course.uuid \
         unless course_event_response.fetch(:is_gap) || course_event_response.fetch(:is_end)
 
+      limit_sequence_number = [
+        course.sequence_number, events.map { |event| event.fetch(:sequence_number) }.max + 1
+      ].max
+
       events_by_type = events.group_by { |event| event.fetch(:event_type) }
```
Comment (nathanstitt) [inline app/domain/services/fetch_course_events/service.rb:138]: you could use `max_by` here https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-max_by

File: app/domain/services/prepare_clue_calculations/service.rb [hunk 2]
```diff
@@ -498,63 +498,84 @@ def process
...
+          updated_scc_uuids = StudentClueCalculation.joins(
+            <<~JOIN_SQL
+              INNER JOIN (#{ValuesTable.new(student_clue_calculation_values)}) AS "values"
+                ("student_uuid", "book_container_uuid")
+                ON "student_clue_calculations"."student_uuid" = "values"."student_uuid"::uuid
+                  AND "student_clue_calculations"."book_container_uuid" =
+                    "values"."book_container_uuid"::uuid
+            JOIN_SQL
+          ).ordered.lock('FOR UPDATE').pluck(:uuid)
+
```
Comment (nathanstitt) [inline app/domain/services/prepare_clue_calculations/service.rb:517]: does the `lock` do anything here?  I thought that for the lock to be effective you needed to use the same ActiveRecord object for the update later.

regardless, I don't see a later update but might be missing it.

**Review by nathanstitt — APPROVED** [review]

File: app/domain/services/fetch_course_events/service.rb (hunk 1 above)
Comment (Dantemss) [reply app/domain/services/fetch_course_events/service.rb:138]: I would have to call fetch twice with `max_by`... I think map/max is a little more concise here.
One interesting alternative would be `([ course.sequence_number ] + events.map { |event| event.fetch(:sequence_number) + 1 }).max`... hmm maybe I'll do that...

File: app/domain/services/prepare_clue_calculations/service.rb (hunk 2 above)
Comment (Dantemss) [reply app/domain/services/prepare_clue_calculations/service.rb:517]: The import right below is the update (upsert).

**Pull request merged**